modules/aws/eks/variables.tf

```diff
@@ -32,16 +32,30 @@ variable "eks_authorized_ips" {
 variable "eks_config" {
   description = "The EKS Config"
   type = object({
-    kubernetes_version = string
-    cidr_block         = string
+    version    = string
+    cidr_block = string
     node_groups = list(object({
-      name            = string
-      release_version = string
-      min_size        = number
-      max_size        = number
-      instance_types  = list(string)
+      name           = string
+      version        = string
+      min_size       = number
+      max_size       = number
+      instance_types = list(string)
     }))
   })
+
+  validation {
+    condition = alltrue([
+      for np in concat(var.eks_config.node_groups, [{ version : var.eks_config.version }]) : can(regex("^1.(19|20|21)", np.version))
+    ])
+    error_message = "The Kubernetes version has not been validated yet, supported versions are 1.19, 1.20, 1.21."
```

Comment:
Sadly it is not possible to use locals in validation so this needs to be hardcoded.